pkg/util/mutate/mutate.go

```diff
@@ -117,66 +143,66 @@ func WithRetry(ctx context.Context, c client.Client, obj client.Object, mutate F
 			// No change necessary.
 			return false, nil
 		}
+		err := c.WrapError(ctx, obj, retryErr)
 		if statusErr := apierrors.APIStatus(nil); errors.As(retryErr, &statusErr) {
 			// The error (or an error it wraps) is a known status error from K8s
-			return false, status.APIServerError(retryErr, "failed to update object", obj)
+			return false, status.APIServerErrorWrap(err, obj)
 		}
-		return false, errors.Wrapf(retryErr, "failed to update object: %s", objSummary(obj))
+		return false, err
 	}
 	return true, nil
 }
 
-// Status attempts to update an object's status, once.
-//
-//   - If the update errors due to a UID conflict, that means the object has
-//     been deleted and re-created. So an error will be returned.
-//   - If the mutate.Func returns a *NoUpdateError, the update will be skipped.
-func Status(ctx context.Context, c client.Client, obj client.Object, mutate Func) (bool, error) {
-	// UID must be set already, so we can error if it changes.
-	uid := obj.GetUID()
-	if uid == "" {
-		return false, fmt.Errorf("failed to update object: %s: metadata.uid is empty", objSummary(obj))
-	}
-	// Wrap inner part in a closure to simplify consistent error handling.
-	// Errors from `Status`` and `WithRetry` should look similar, even tho
-	// `Status` only makes one attempt.
-	retryErr := func() error {
-		// If ResourceVersion is NOT set, request the latest object from the server.
-		// Otherwise, assume the object was previously retrieved.
-		if obj.GetResourceVersion() == "" {
-			objKey := client.ObjectKeyFromObject(obj)
-			if getErr := c.Get(ctx, objKey, obj); getErr != nil {
-				return errors.Wrap(getErr, "failed to get latest version of object")
-			}
-			if obj.GetUID() != uid {
-				return errors.New("metadata.uid has changed: object may have been re-created")
-			}
-		}
-		if err := mutate(); err != nil {
-			return errors.Wrap(err, "failed to mutate object")
-		}
-		return c.Status().Update(ctx, obj)
-	}()
-	if retryErr != nil {
-		if noUpdateErr := (&NoUpdateError{}); errors.As(retryErr, &noUpdateErr) {
-			// No change necessary.
-			return false, nil
-		}
-		if statusErr := apierrors.APIStatus(nil); errors.As(retryErr, &statusErr) {
-			// The error (or an error it wraps) is a known status error from K8s
-			return false, status.APIServerError(retryErr, "failed to update object status", obj)
-		}
-		return false, errors.Wrapf(retryErr, "failed to update object status: %s", objSummary(obj))
-	}
-	return true, nil
+// updater is a simplified client interface for performing reads and writes on
+// an object, without options.
+type updater interface {
+	// Get the current state of the specified object.
```

Comment:
`Get the current state of the specified object.` to `Get the specified object.` since there are two implementations of this interface: `specClient` and `statusClient`.

Reply:
done